### ISSUE TITLE
RabbitMQ ERROR: inequivalent arg 'x-queue-type' for queue 'queue-order' in vhost '/': received none but current is the value 'classic' of type 'longstr'

### ISSUE THREAD

**Issue by AmosWong1998** [spring-projects/spring-amqp#1150]
This error is always reported when I start the app.
### version

```java
SpringBoot 2.2.2.RELEASE
RabbitMQ 3.8.1.RELEASE
```

### erorr logs:

```java
2020-01-29 18:53:32.752 ERROR 19484 --- [ 127.0.0.1:5672] o.s.a.r.c.CachingConnectionFactory       : Channel shutdown: channel error; protocol method: #method<channel.close>(reply-code=406, reply-text=PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'queue-order' in vhost '/': received none but current is the value 'classic' of type 'longstr', class-id=50, method-id=10)
2020-01-29 18:53:33.756 ERROR 19484 --- [ 127.0.0.1:5672] o.s.a.r.c.CachingConnectionFactory       : Channel shutdown: channel error; protocol method: #method<channel.close>(reply-code=406, reply-text=PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'queue-order' in vhost '/': received none but current is the value 'classic' of type 'longstr', class-id=50, method-id=10)
2020-01-29 18:53:35.759 ERROR 19484 --- [ 127.0.0.1:5672] o.s.a.r.c.CachingConnectionFactory       : Channel shutdown: channel error; protocol method: #method<channel.close>(reply-code=406, reply-text=PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'queue-order' in vhost '/': received none but current is the value 'classic' of type 'longstr', class-id=50, method-id=10)
```

### application.yml
```java
spring:
  rabbitmq:
    host: 127.0.0.1
    port: 5672
    username: guest
    password: guest
    virtual-host: /
    listener:
      simple:
        prefetch: 10
        acknowledge-mode: manual 
```

### my consumer service:

```java
@RabbitListener(
            bindings = @QueueBinding(
                    value = @Queue(value = "queue-order", durable = "true"),
                    exchange = @Exchange(value = "exchange-order", type = "fanout")
            )
    )
    @RabbitHandler
    @Transactional
    public void handleMessage(@Payload Map<String, Object> data, Channel channel, @Headers Map<String, Object> headers) {
        try {
           //ignore...
            Long tag = (Long) headers.get(AmqpHeaders.DELIVERY_TAG);
            channel.basicAck(tag, false);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
```

### According to the following error report, I guess it may be the problem of the attribute 'x-queue-type' 

```java
inequivalent arg 'x-queue-type' for queue 'queue-order' in vhost '/': received none but current is the value 'classic' of type 'longstr'
```
So I change my consumer service:

```java
@RabbitListener(
            bindings = @QueueBinding(
                    value = @Queue(value = "queue-order", durable = "true"),
                    exchange = @Exchange(value = "exchange-order", type = "fanout"),
                   // add arguments attribute
                    arguments = {
                            @Argument(name = "x-queue-type", value = "classic")
                    }
            )
    )
```
But it doesn't work. It still reports the same error

**Comment by AmosWong1998** [comment]
oh, I find my problem, the consumer service should be:
```java
@RabbitListener(
            bindings = @QueueBinding(
                    value = @Queue(value = "queue-order", durable = "true", arguments = {
                            @Argument(name = "x-queue-type", value = "classic")
                    }),
                    exchange = @Exchange(value = "exchange-order", type = "fanout")
            )
    )
```
but why throw this error？？

**Comment by garyrussell** [comment]
You should ask questions about RabbitMQ itself on the rabbitmq-users Google Group.

This was a bug in RabbitMQ itself; [fixed in 3.8.2](https://github.com/rabbitmq/rabbitmq-common/issues/341).

**Comment by AmosWong1998** [comment]
Okey, thank you so much. @garyrussell
